yggdrasil/services/monitoring/tempo/tempo.yaml

```
@@ -0,0 +1,246 @@
+bucketClaim:
+  # Enableing the bucket claim
+  enabled: true
+  # The name of the bucket claim
+  name: tempo-bucket-claim
+  # Annotations for the bucket claim
+  annotations:
+    # This annotations makes it so ArgoCD will not complain if the CRD's are missing
+    argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
+  # Name of the bucket in which to make the claim
+  bucketName: tempo-bucket
+  # Name of the storageClass that contains the bucket provider
+  storageClassName: ceph-bucket
+
+tempo-distributed:
+  ingester:
+    # `config.expand-env: "true"` Allows the use of env vars in the config
+    extraArgs:
+      - -config.expand-env=true
+
+    # -- Environment variables to add to the ingester pods
+    extraEnv:
+      - name: AWS_ACCESS_KEY_ID
+        valueFrom:
+          secretKeyRef:
+            name: tempo-bucket-claim
+            key: AWS_ACCESS_KEY_ID
+      - name: AWS_SECRET_ACCESS_KEY
+        valueFrom:
+          secretKeyRef:
+            name: tempo-bucket-claim
+            key: AWS_SECRET_ACCESS_KEY
+      - name: BUCKET_HOST
+        valueFrom:
+          configMapKeyRef:
+            name: tempo-bucket-claim
+            key: BUCKET_HOST
+      - name: BUCKET_NAME
+        valueFrom:
+          configMapKeyRef:
+            name: tempo-bucket-claim
+            key: BUCKET_NAME
+      - name: BUCKET_REGION
+        valueFrom:
+          configMapKeyRef:
+            name: tempo-bucket-claim
+            key: BUCKET_REGION
```

Can't we set them in `config`?

https://github.com/grafana/helm-charts/tree/652c984229db23edd62446447ebfb2d997f5d0e5/charts/tempo-distributed#example-configuration-using-s3-for-storage

---

We are just [here](https://github.com/distributed-technologies/yggdrasil/blob/932aa0dea18f2593c0cdb3a4179dd5a7927adfdd/yggdrasil/services/monitoring/tempo/tempo.yaml#L183-L212) instead, since you still have to mount the variables to the pod that's using it in their configs.

This code just mounts the values from the configMap and secret as env variables on the pod

---

Can we do:
```suggestion
    extraEnv:
      envFrom:
      - secretRef:
          name: tempo-bucket-claim
      - configMapRef:
          name: tempo-bucket-claim
```
and perhaps move it `_helpers.tpl`, so we can define the section once and reuse?

---

That could work in tempo, since they have value called `extraEnvFrom` [here](https://github.com/grafana/helm-charts/blob/652c984229db23edd62446447ebfb2d997f5d0e5/charts/tempo-distributed/templates/distributor/deployment-distributor.yaml#L95-L99).

This will not work in cortex since they've only mapped the env value [here](https://github.com/cortexproject/cortex-helm-chart/blob/ad2f5cf3ce961af813fc73b9530be75a585322ac/templates/distributor/distributor-dep.yaml#L93) 

Plus I prefer to explicit state what env variables I want.

I had not thought about moving it to a _helper.tpl, which would make it a lot neater, I would like to do that, but keep the explicit value declaration, thoughts?